app/listeners/base_listener.rb

```diff
@@ -22,4 +22,10 @@ def extract_contact_and_account(event)
     contact = event.data[:contact]
     [contact, contact.account]
   end
+
+  def extract_changed_attributes(event)
+    changed_attributes = event.data[:changed_attributes]
+    changed_attributes.map { |k, v| { k => { previous_value: v[0], current_value: v[1] } } }
```

Comment:
Made this change to handle https://github.com/chatwoot/chatwoot/pull/4317#issuecomment-1081463383